[R2] Add R2 terraform example using cloudflare provider (#9103)
cloudflare/cloudflare-docs: content/r2/examples/terraform.md

{{<render file="_keys.md">}}<br>

> This example shows how to configure R2 with Terraform using the [Cloudflare provider](https://github.com/cloudflare/terraform-provider-cloudflare).The For more advanced use cases, see.

Reviewer: ```suggestion
This example shows how to configure R2 with Terraform using the [Cloudflare provider](https://github.com/cloudflare/terraform-provider-cloudflare).
```

{{<Aside type="note" header="Note for using AWS provider">}}

> When using the Cloudflare Terraform provider, you are only able to manage buckets. For configuring items such as CORS and lifecycle, you will need to use the [AWS Provider](/r2/examples/terraform-aws/).

Reviewer: ```suggestion
When using the Cloudflare Terraform provider, you can only manage buckets. To configure items such as CORS and object lifecycles, you will need to use the [AWS Provider](/r2/examples/terraform-aws/).
```
